neuromation/api/tracing.py

```diff
@@ -0,0 +1,42 @@
+# Distributed tracing support
+
+import binascii
+import os
+import types
+
+import aiohttp
+from multidict import CIMultiDict
+
+
+def gen_trace_id() -> str:
+    """Returns a 64 bit UTF-8 encoded string.
+
+    The id is used for distributed tracing.
+    """
+    return str(binascii.hexlify(os.urandom(8)).decode("utf-8"))
+
+
+def _update_headers(headers: "CIMultiDict[str]", trace_id: str, span_id: str) -> None:
+    """Creates dict with zipkin single header format.
+    """
+    # b3={TraceId}-{SpanId}-{SamplingState}-{ParentSpanId}
```

Comment:
It works only if this parent span is stored in DB.
Since we don't save to zipkin from a client the span doesn't exist.
Recursive upload is still grouped together because everything shares the same trace id.
I believe it should be enough.
Server code *will* use parent span id feature, e.g. for requesting auth microservice.

Reply:
interesting. I might be missing smith. I was thinking about the following use case:
we want to download a directory. first, we get a file listing, then we spawn tasks and pull the data based on the listing. is this also a legit case for leveraging parent spans?